### PR TITLE
Methods for getting IP addresses (v4 or v6)

### PR DESCRIPTION
**Pull request by raboof**
For code review, but not for merge yet, as I'm not currently at an IPv6-connected machine so I can't really test.

Also still needs a test in inetdiag_test.go

### CODE REVIEW

**Comment by zhexuany** [comment]
@raboof Thanks for the PR. I have a question for you. We already have two method call `SrcIPv4()` and `DstIPv4()`, why do't you just use these two? Additionally, you are not in a `IPv6` environment. I do not really see any point for adding `IPv6` support.

**Comment by raboof** [comment]
Thanks for your quick response. While the machine I'm currently working on is not IPv6-connected, the software I'm writing (https://github.com/raboof/connbeat) will be deployed on IPv6-connected machines in the future. In that situation `SrcIPv4` and `DstIPv4` of course will not be sufficient.

**Comment by zhexuany** [comment]
@thbourlove It seems like, we have enough reason to support `IPv6`. What is your opinion? Since we use this `netlink`internally a lot, we can't just remove old `SrcIPv4()` and `DstIPv4()`. My workaround solution is that we create two new APIs `SrcIPv6()` and `DstIPv6()`. After we finished clean our own codebase, we can wrap these API with `SrcIP()` and `DstIP()`.

File: inetdiag.go [hunk 1]
```diff
@@ -78,6 +78,54 @@ func (id *InetDiagSockId) DstIPv4() net.IP {
...
+func ipv6ip(original [4]be32) net.IP {
+	isIpv6 := false
+	for i := 1; i < 4; i++ {
+		for j := 0; j < 4; j++ {
+			isIpv6 = true
+		}
+	}
+
+
+	if isIpv6 {
```
Comment (zhexuany) [inline inetdiag.go]: The logic here should be negated.

**Comment by thbourlove** [comment]
👍

**Pull request merged**